common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidKeystoreAsymmetricKey.java

```diff
@@ -0,0 +1,107 @@
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.platform;
+
+import android.content.Context;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.microsoft.identity.common.exception.ClientException;
+
+import java.util.Date;
+
+/**
+ * Represents an asymmetric key, backed by the Android Keystore.
+ */
+public class AndroidKeystoreAsymmetricKey implements AsymmetricKey {
+
+    /**
+     * The {@link IDevicePopManager} to which we will delegate most cryptographic actions.
+     */
+    private final IDevicePopManager mDevicePopManager;
+
+    /**
+     * The alias is name of this keypair in the underlying keystore.
+     */
+    private final String mAlias;
+
+    /**
+     * Constructs a new {@link AndroidKeystoreAsymmetricKey} instance.
+     *
+     * @param context    The application Context.
+     * @param popManager The underlying {@link IDevicePopManager} to which we'll delegate.
+     * @throws ClientException If asymmetric key generation fails.
+     */
+    AndroidKeystoreAsymmetricKey(@NonNull final Context context,
+                                 @NonNull final IDevicePopManager popManager,
+                                 @NonNull final String alias)
+            throws ClientException {
+        mDevicePopManager = popManager;
+        mAlias = alias;
+
+        if (!mDevicePopManager.asymmetricKeyExists()) {
+            mDevicePopManager.generateAsymmetricKey(context);
+        }
+    }
+
+    @Override
+    public String getAlias() {
+        return mAlias;
+    }
+
+    @Override
+    public Date getCreatedOn() throws ClientException {
+        return mDevicePopManager.getAsymmetricKeyCreationDate();
+    }
+
+    @Override
+    public String getThumbprint() throws ClientException {
+        return mDevicePopManager.getAsymmetricKeyThumbprint();
+    }
+
+    @Override
+    public String getPublicKey() throws ClientException {
+        return mDevicePopManager.getPublicKey(IDevicePopManager.PublicKeyFormat.JWK);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
+    @Override
+    public String sign(@NonNull final String data) throws ClientException {
+        return mDevicePopManager.sign(
+                DevicePopManager.SigningAlgorithm.SHA_256_WITH_RSA,
+                data
+        );
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
+    @Override
+    public boolean verify(@NonNull final String plainText, @NonNull final String signatureStr) {
+        return mDevicePopManager.verify(
+                DevicePopManager.SigningAlgorithm.SHA_256_WITH_RSA,
```

Comment:
If this is coupled to RSA, I'd incorporate that into the implementing class name.

Reply:
ACK -- will be part of the name

Reply:
If we do, I'd go ahead and make it a subinterface, keeping the superinterface generic w.r.t. algorithm.  I'd hate to add too much friction to an eventual move away from RSA.

Reply:
Pulled up in 32e3fc7df261f5b9660f42e683b476e0bb38dff5